File: sdks/ts/src/sdk/models/operations/putgroupmembershipsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PutGroupMembershipsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=group_membership_id" })
  groupMembershipId: string;
}


export class PutGroupMembershipsIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}

export enum PutGroupMembershipsIdRequestBodyRoleEnum {
    Member = "member",
    Admin = "admin"
}


export class PutGroupMembershipsIdRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=configurable_permissions" })
  configurablePermissions?: Record<string, boolean>;

  @SpeakeasyMetadata({ data: "json, name=role" })
  role?: PutGroupMembershipsIdRequestBodyRoleEnum;
}


export class PutGroupMembershipsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PutGroupMembershipsIdPathParams;

  @SpeakeasyMetadata()
  queryParams: PutGroupMembershipsIdQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PutGroupMembershipsIdRequestBody;
}


export class PutGroupMembershipsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  groupMembership?: shared.GroupMembership;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/putgroupmembershipsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PutGroupMembershipsIDPathParams struct {
	GroupMembershipID string `pathParam:"style=simple,explode=false,name=group_membership_id"`
}

type PutGroupMembershipsIDQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type PutGroupMembershipsIDRequestBodyRoleEnum string

const (
	PutGroupMembershipsIDRequestBodyRoleEnumMember PutGroupMembershipsIDRequestBodyRoleEnum = "member"
	PutGroupMembershipsIDRequestBodyRoleEnumAdmin  PutGroupMembershipsIDRequestBodyRoleEnum = "admin"
)

type PutGroupMembershipsIDRequestBody struct {
	ConfigurablePermissions map[string]bool                           `json:"configurable_permissions,omitempty"`
	Role                    *PutGroupMembershipsIDRequestBodyRoleEnum `json:"role,omitempty"`
}

type PutGroupMembershipsIDRequest struct {
	Retries     *utils.RetryConfig
	PathParams  PutGroupMembershipsIDPathParams
	QueryParams PutGroupMembershipsIDQueryParams
	Request     *PutGroupMembershipsIDRequestBody `request:"mediaType=application/json"`
}

type PutGroupMembershipsIDResponse struct {
	ClientError     *shared.ClientError
	ContentType     string
	GroupMembership *shared.GroupMembership
	StatusCode      int64
}
